src/job_cache/types.cpp

```diff
@@ -107,62 +107,112 @@ JAST JobOutputInfo::to_json() const {
   return json;
 }
 
-MatchingJob::MatchingJob(const JAST &json) {
+MatchingJob::MatchingJob(const std::string &client_cwd, const JAST &json) {
   output_info = JobOutputInfo(json.get("output_info"));
 
   for (const auto &output_file_json : json.get("output_files").children) {
-    output_files.push_back(CachedOutputFile(output_file_json.second));
+    CachedOutputFile output_file(output_file_json.second);
+    // Canoncolize matching jobs to use client-relative paths.
```

Comment:
sp: Canonicalize

Reply:
Ah that was also something I realized was wrong mid-writing. Comment removed!